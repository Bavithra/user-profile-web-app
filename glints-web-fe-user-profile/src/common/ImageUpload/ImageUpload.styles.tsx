import styled from "styled-components";
import { Colors } from "../../styles/Colors";

export const ImageContainer = styled.div`
  display: flex;
  flex-direction: column;
    align-items: center;
`;

export const Image = styled.img`
  border-radius: 50%;
  border: ${Colors.grey1} 2px solid;
  object-fit: cover;
  width: 200px;
  height: 200px;
`;

export const FileInput = styled.input`
  display: none;
`;
