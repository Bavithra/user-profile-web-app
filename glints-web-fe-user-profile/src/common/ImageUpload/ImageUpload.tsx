import React, { ChangeEvent, useRef } from "react";

import { FileInput, Image, ImageContainer } from "./ImageUpload.styles";

import Avatar from "../../assets/avatar.png";
import Button from "../Button";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPlus } from '@fortawesome/free-solid-svg-icons';

export default function ImageUpload() {
  const [fileSelected, setFileSelected] = React.useState<
    string | ArrayBuffer | null
  >();

  const fileInputRef = useRef<HTMLInputElement>(null);

  function handleButtonClick() {
    fileInputRef.current && fileInputRef.current.click();
  }

  function handleInputChange(event: ChangeEvent<HTMLInputElement>) {
    if (event.target.files === null || event.target.files.length === 0) {
      return;
    }

    const file = event.target.files[0];

    const reader = new FileReader();

    reader.onload = function (e) {
      if (e === null || e.target === null) {
        return;
      }

      setFileSelected(URL.createObjectURL(file));
    };

    reader.readAsText(file);
  }

  return (
    <ImageContainer>
      <Image alt="" src={fileSelected ? fileSelected.toString() : Avatar} />
      <Button onClick={handleButtonClick}><FontAwesomeIcon icon={faPlus} size="xs" color={'white'} /> Upload Image</Button>
      <FileInput
        data-testid="file-input"
        ref={fileInputRef}
        type="file"
        accept="image/*"
        onChange={handleInputChange}
      />
    </ImageContainer>
  );
}
