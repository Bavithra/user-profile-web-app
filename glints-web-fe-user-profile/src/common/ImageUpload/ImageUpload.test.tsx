import React from "react";
import ImageUpload from "./ImageUpload";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";

describe("ImageUpload Block", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ImageUpload />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders upload image button", () => {
    render(<ImageUpload />);
    const buttonElement = screen.getByText("Upload Image");
    expect(buttonElement).toBeInTheDocument();
  });
});
