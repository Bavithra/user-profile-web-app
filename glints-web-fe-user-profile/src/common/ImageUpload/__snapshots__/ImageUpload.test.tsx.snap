// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageUpload Block renders correctly 1`] = `
<div
  className="sc-bczRLJ hZeqla"
>
  <img
    alt=""
    className="sc-gsnTZi eioidg"
    src="avatar.png"
  />
  <button
    className="sc-hKMtZM dnvXTe"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-plus fa-xs "
      color="white"
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M432 256c0 17.69-14.33 32.01-32 32.01H256v144c0 17.69-14.33 31.99-32 31.99s-32-14.3-32-31.99v-144H48c-17.67 0-32-14.32-32-32.01s14.33-31.99 32-31.99H192v-144c0-17.69 14.33-32.01 32-32.01s32 14.32 32 32.01v144h144C417.7 224 432 238.3 432 256z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
     Upload Image
  </button>
  <input
    accept="image/*"
    className="sc-dkzDqf tPwAK"
    data-testid="file-input"
    onChange={[Function]}
    type="file"
  />
</div>
`;
